cfme/containers/provider/openshift.py

```diff
@@ -63,11 +83,45 @@ class OpenshiftProvider(ContainersProvider):
     db_types = ["Openshift::ContainerManager"]
     endpoints_form = ContainersProviderEndpointsForm
 
-    def __init__(self, name=None, key=None, zone=None, metrics_type=None,
-                 provider_data=None, endpoints=None, appliance=None):
+    def __init__(
+            self,
+            name=None,
+            key=None,
+            zone=None,
+            metrics_type=None,
+            alerts_type=None,
+            provider_data=None,
+            endpoints=None,
+            appliance=None,
+            http_proxy=None,
+            adv_http=None,
+            adv_https=None,
+            no_proxy=None,
+            image_repo=None,
+            image_reg=None,
+            image_tag=None,
+            cve_loc=None):
+
+        gt_59 = True if self.appliance.version >= '5.9' else False
+
+        self.http_proxy = http_proxy if gt_59 else None
```

Comment:
I don't like that setting of those attributes depends on which appliance version is being used. It just isn't a behaviour I would expect. Usually the initialiser function (__init__) doesn't contain such logic. It ususaly just is blindly setting the state of the object.

Reply:
What more, instead of just ignoring the arg, you should at least log some warning or raise exception. Silent ignoration is often bad for people trying to debug.

Reply:
Dicussing with @mshriver where to move it to. It use to be in from_config but I was asked to move it from there.

Reply:
Moved to view_value_mapping